src/main/java/com/hardvan/springboot/springolearn/controller/BookController.java

```diff
@@ -35,4 +39,22 @@ public String deleteAllReviews(@RequestParam String title) {
         bookService.deleteAllReviews(title);
         return "redirect:/";
     }
+
+    @GetMapping("/search")
+    public String searchBooks(@RequestParam String query, @RequestParam String searchType, Model model) {
```

Comment:
You can add function documentation comments here with the help of [javadoc](https://www.oracle.com/in/technical-resources/articles/java/javadoc-tool.html) to make the function's functionality more readable.

Reply:
Oh, okay.